### PR TITLE
added docker

### PR DESCRIPTION
**Pull request by keda-vlad**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+LABEL authors="Влад"
+# Final stage
+FROM openjdk:17-jdk-slim
+WORKDIR application
+COPY --from=builder application/dependencies/ ./
+COPY --from=builder application/spring-boot-loader/ ./
+COPY --from=builder application/snapshot-dependencies/ ./
+COPY --from=builder application/application/ ./
+ENTRYPOINT ["java", "org.springframework.boot.loader.JarLauncher"]
+EXPOSE 8080
```
Comment (vovasalyha) [inline Dockerfile]: empty line

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+version: "3.8"
+
+services:
```
Comment (vovasalyha) [inline docker-compose.yaml:3]: add also service to run the backend app from the Dockerfile

File: .env [hunk 3]
```diff
@@ -0,0 +1,2 @@
+MYSQL_ROOT_PASSWORD=Born_to_sql1
+MYSQL_DATABASE=book_store
```
Comment (vovasalyha) [inline .env]: empty line

File: .gitignore [hunk 4]
```diff
@@ -15,6 +15,7 @@ target/
 
 ### IntelliJ IDEA ###
 .idea
+.env
```
Comment (vovasalyha) [inline .gitignore]: It's OK to push .env file to git so others can test your project easily

File: pom.xml [hunk 5]
```diff
@@ -77,6 +77,14 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-security</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-docker-compose</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.postgresql</groupId>
```
Comment (vovasalyha) [inline pom.xml]: Where do you use this driver?

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Good, but let's improve it a bit

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    image: bookstore-app
+    env_file: ./.env
+    ports:
+      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
+      - $DEBUG_PORT
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:mysql://mysqldb:$MYSQL_DOCKER_PORT/$MYSQL_DB
+      SPRING_DATASOURCE_USERNAME: $MYSQL_USER
+      SPRING_DATASOURCE_PASSWORD: $MYSQL_PASSWORD
+      JAVA_TOOL_OPTIONS: -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:$DEBUG_PORT
```
Comment (nacenik) [inline docker-compose.yaml]: add empty line

**Review by nacenik — CHANGES_REQUESTED** [review]
Fix pr history. You have a lot of changes that are not related to the task.

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**